pkgs/development/compilers/symbiflow-yosys/default.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv
+, abc-verifier
+, bash
+, bison
+, fetchFromGitHub
+, flex
+, libffi
+, pkgconfig
+, protobuf
+, python3
+, readline
+, tcl
+, verilog
+, zlib
+, plugins ? []
+}:
+
+let
+  localAbc-verifier = abc-verifier.overrideAttrs (_: rec {
+    pname = "abc-verifier";
+    version = "2020.06.22";
+
+    src = fetchFromGitHub {
+      owner  = "YosysHQ";
+      repo   = "abc";
+      rev    = "341db25668f3054c87aa3372c794e180f629af5d";
+      sha256 = "14cgv34vz5ljkcms6nrv19vqws2hs8bgjgffk5q03cbxnm2jxv5s";
+    };
+
+    passthru.rev = src.rev;
+  });
+in
+
+stdenv.mkDerivation rec {
```

Comment:
Why can't we just do
```nix
yosys.overrideAttrs (_: {
  # ...
```

Reply:
We'd have to override, besides src: makeFlags, postInstall and preBuild.

Reply:
Would that not be better than separately maintain `symbiflow-yosys` and `yosys`?

Reply:
Like this: ```nix
( yosys.overrideAttrs (oldAttrs: rec {
  inherit plugins;

  pname   = "symbiflow-yosys";
  version = "0.9+2406";

  src = fetchFromGitHub {
    owner  = "SymbiFlow";
    repo   = "yosys";
    rev    = "d8b2d1a2b1a93057678cf49bb8f0329f191faba1";
    sha256 = "1w8jnqzabvzixjllhb6ak2n2gmjvsn6qd996i7z70bsq5rgdkq9g";
  };

  postInstall = ''
    ln -sfv ${abc-verifier}/bin/abc $out/bin/yosys-abc

    mkdir -p $out/share/yosys/plugins
    for plugin in $plugins; do
        ln -sfv $plugin/share/yosys/plugins/* $out/share/yosys/plugins/
    done
  '';

  preBuild = builtins.replaceStrings [ oldAttrs.version ] [ version ] oldAttrs.preBuild;
})).override {
  abc-verifier = localAbc-verifier;
}
```

Reply:
Could add plugin functionality to `yosys` too. Might as well make version overridable:
```diff
diff --git a/pkgs/development/compilers/yosys/default.nix b/pkgs/development/compilers/yosys/default.nix
index 51f0b7ed45a..a67632aa554 100644
--- a/pkgs/development/compilers/yosys/default.nix
+++ b/pkgs/development/compilers/yosys/default.nix
@@ -76,8 +76,8 @@ stdenv.mkDerivation rec {
       exit 1
     fi
 
-    if ! grep -q "YOSYS_VER := ${version}" Makefile; then
-      echo "ERROR: yosys version in Makefile isn't equivalent to version of the nix package (${version}), failing."
+    if ! grep -q "YOSYS_VER := $version" Makefile; then
+      echo "ERROR: yosys version in Makefile isn't equivalent to version of the nix package ($version), failing."
       exit 1
     fi
   '';
```